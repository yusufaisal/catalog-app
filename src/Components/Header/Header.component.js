import * as React from 'react';
import { View, Text, TouchableOpacity } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

import styles from './Header.component.styles';

const _renderBackButton = () => (
  <TouchableOpacity style={styles.backButton}>
    <Icon size={24} name="arrow-back-outline" color="#080808" />
  </TouchableOpacity>
);

const _renderFilterButton = () => (
  <TouchableOpacity style={styles.filterButton}>
    <Icon size={24} name="options-outline" color="#080808" />
  </TouchableOpacity>
);

const _renderTitle = (props) => (
  <Text style={styles.title}>{props.title}</Text>
);

const Header = (props) => (
  <View style={styles.container}>
    {_renderBackButton()}
    {_renderTitle(props)}
    {_renderFilterButton()}
  </View>
)

export default Header;