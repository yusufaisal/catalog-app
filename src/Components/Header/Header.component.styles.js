import { StyleSheet } from 'react-native';

const shadow = {
  shadowColor: '#171717',
  shadowOffset: {width: 1, height: 1},
  shadowOpacity: 0.1,
  shadowRadius: 3,
};

export default StyleSheet.create({
  container: {
    flexDirection: 'row',
    height: 125,
    paddingTop: 50,
    paddingBottom: 20,
    paddingHorizontal: 20,
    backgroundColor: '#F5F5F5',
    alignItems: 'flex-end'
  },
  title: {
    fontSize: 24,
    color: '#110F2B',
    fontWeight: 'bold',
    paddingBottom: 4
  },
  backButton: {
    flex: 0,
    height: 42,
    width: 42,
    marginRight: 20,
    borderRadius: 21,
    backgroundColor: '#FFFFFF',
    justifyContent: 'center',
    alignItems: 'center',
    ...shadow
  },
  filterButton: {
    position: 'absolute',
    height: 42,
    width: 42,
    right: 0,
    bottom: 20,
    marginRight: 20,
    borderRadius: 21,
    backgroundColor: '#FFFFFF',
    justifyContent: 'center',
    alignItems: 'center',
    ...shadow
  }
});