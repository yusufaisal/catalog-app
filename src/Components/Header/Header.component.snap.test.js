import React from 'react';
import renderer from 'react-test-renderer';

import Header from './Header.component';

jest
  .mock('react-native-vector-icons/Ionicons', () => 'Icon');
  
describe('Header Component', () => {
  it('should render Header properly', () => {
    const tree = renderer
      .create(<Header title={'Shoes'}/>)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});