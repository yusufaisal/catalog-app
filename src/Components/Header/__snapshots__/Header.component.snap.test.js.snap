// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header Component should render Header properly 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "backgroundColor": "#F5F5F5",
      "flexDirection": "row",
      "height": 125,
      "paddingBottom": 20,
      "paddingHorizontal": 20,
      "paddingTop": 50,
    }
  }
>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 21,
        "flex": 0,
        "height": 42,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "shadowColor": "#171717",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 3,
        "width": 42,
      }
    }
  >
    <Icon
      color="#080808"
      name="arrow-back-outline"
      size={24}
    />
  </View>
  <Text
    style={
      Object {
        "color": "#110F2B",
        "fontSize": 24,
        "fontWeight": "bold",
        "paddingBottom": 4,
      }
    }
  >
    Shoes
  </Text>
  <View
    accessible={true}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#FFFFFF",
        "borderRadius": 21,
        "bottom": 20,
        "height": 42,
        "justifyContent": "center",
        "marginRight": 20,
        "opacity": 1,
        "position": "absolute",
        "right": 0,
        "shadowColor": "#171717",
        "shadowOffset": Object {
          "height": 1,
          "width": 1,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 3,
        "width": 42,
      }
    }
  >
    <Icon
      color="#080808"
      name="options-outline"
      size={24}
    />
  </View>
</View>
`;
